cp_core/src/task/transform/common.rs

```diff
@@ -90,25 +90,76 @@ impl Stage for RootTransform {
 }
 
 impl Transform for RootTransform {
-    /// Runs execution in order. ctx is passed to children stages
+    /// Runs execution in order. ctx is passed to children subtransforms
     /// which can also extract changes on frames
     fn run(&self, main: LazyFrame, ctx: Arc<DefaultPipelineContext>) -> CpResult<LazyFrame> {
         let mut next = main;
-        for stage in &self.stages {
+        for stage in &self.subtransforms {
             next = stage.as_ref().run(next, ctx.clone())?
         }
         Ok(next)
     }
 }
 
-impl SubTransformConfig for RootTransformConfig {
-    fn transform(self) -> Box<dyn Transform> {
-        // YX TODO
-        todo!()
+macro_rules! try_deserialize {
+    ($value:expr, $($type:ty),+) => {
+        $(if let Ok(config) = serde_yaml_ng::from_value::<$type>($value.clone()) {
+            Some(Box::new(config) as Box<dyn TransformConfig>)
+        }) else+
+        else {
+            None
+        }
+    };
+}
+
+impl RootTransformConfig {
+    fn parse_subtransforms(&self) -> Vec<Result<Box<dyn TransformConfig>, CpError>> {
+        self.subtransforms
+            .iter()
+            .map(|transform| {
+                let config = try_deserialize!(transform, SelectTransformConfig, JoinTransformConfig);
+
+                config.ok_or_else(|| {
+                    CpError::ConfigError(
+                        "Transform config parsing error",
+                        format!("Failed to parse transform config: {:?}", transform),
+                    )
+                })
+            })
+            .collect()
     }
+}
+
+impl TransformConfig for RootTransformConfig {
     fn validate(&self) -> Vec<CpError> {
-        // YX TODO
-        todo!()
+        let mut errors = vec![];
+
+        for result in self.parse_subtransforms() {
+            match result {
+                Ok(config) => {
+                    errors.extend(config.validate().into_iter().map(|e| {
+                        CpError::ConfigError("Transform config validation error", format!("Subtransform: {}", e))
+                    }));
```

Comment:
```suggestion
                    errors.extend(config.validate());
```

I think there is no need for the additional map, because each config's `validate` call already returns a `Vec<CpError>`